File: software/examples/java/ExampleSetSegments.java
import com.tinkerforge.IPConnection;
import com.tinkerforge.BrickletSegmentDisplay4x7V2;

public class ExampleSetSegments {
	private static final String HOST = "localhost";
	private static final int PORT = 4223;

	// Change XYZ to the UID of your Segment Display 4x7 Bricklet 2.0
	private static final String UID = "XYZ";

	// Note: To make the example code cleaner we do not handle exceptions. Exceptions
	//       you might normally want to catch are described in the documentation
	public static void main(String args[]) throws Exception {
		IPConnection ipcon = new IPConnection(); // Create IP connection
		BrickletSegmentDisplay4x7V2 sd =
		  new BrickletSegmentDisplay4x7V2(UID, ipcon); // Create device object

		ipcon.connect(HOST, PORT); // Connect to brickd
		// Don't use device before ipcon is connected

		sd.setBrightness(7); // Set to full brightness

		// Activate all segments
		sd.setSegments(new boolean[]{true, true, true, true, true, true, true, true},
		               new boolean[]{true, true, true, true, true, true, true, true},
		               new boolean[]{true, true, true, true, true, true, true, true},
		               new boolean[]{true, true, true, true, true, true, true, true},
		               new boolean[]{true, true}, true);

		System.out.println("Press key to exit"); System.in.read();
		ipcon.disconnect();
	}
}

File: software/examples/go/example_set_segments.go
package main

import (
	"fmt"
	"github.com/Tinkerforge/go-api-bindings/ipconnection"
	"github.com/Tinkerforge/go-api-bindings/segment_display_4x7_v2_bricklet"
)

const ADDR string = "localhost:4223"
const UID string = "XYZ" // Change XYZ to the UID of your Segment Display 4x7 Bricklet 2.0.

func main() {
	ipcon := ipconnection.New()
	defer ipcon.Close()
	sd, _ := segment_display_4x7_v2_bricklet.New(UID, &ipcon) // Create device object.

	ipcon.Connect(ADDR) // Connect to brickd.
	defer ipcon.Disconnect()
	// Don't use device before ipcon is connected.

	sd.SetBrightness(7) // Set to full brightness

	// Activate all segments
	sd.SetSegments([8]bool{true, true, true, true, true, true, true, true},
		[8]bool{true, true, true, true, true, true, true, true},
		[8]bool{true, true, true, true, true, true, true, true},
		[8]bool{true, true, true, true, true, true, true, true},
		[2]bool{true, true}, true)

	fmt.Print("Press enter to exit.")
	fmt.Scanln()
}
